From 1857ee3fd7e7386c09ced900538f02909726427e Mon Sep 17 00:00:00 2001
From: Evan Severson <evanseverson@google.com>
Date: Tue, 4 Apr 2023 14:49:26 -0700
Subject: [PATCH] Move startWatchingModeWithFlags to the native supported
 binder calls

Bug: 247768581
Test: Add logging to verify invocation
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:b1f82ee37403e40513ef3b9e2657feb3871c4e71)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:fc85ac068d58f5cc690d5d587a46f6f691b0ce3a)
Merged-In: I54eefca5f0aa4f924debc1817b04a103b6e8e2e6
Change-Id: I54eefca5f0aa4f924debc1817b04a103b6e8e2e6
---
 core/java/com/android/internal/app/IAppOpsService.aidl | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/java/com/android/internal/app/IAppOpsService.aidl b/core/java/com/android/internal/app/IAppOpsService.aidl
index 88447daf7338..ff3c015cf66f 100644
--- a/core/java/com/android/internal/app/IAppOpsService.aidl
+++ b/core/java/com/android/internal/app/IAppOpsService.aidl
@@ -52,6 +52,8 @@ interface IAppOpsService {
     int checkAudioOperation(int code, int usage, int uid, String packageName);
     boolean shouldCollectNotes(int opCode);
     void setCameraAudioRestriction(int mode);
+    void startWatchingModeWithFlags(int op, String packageName, int flags,
+            IAppOpsCallback callback);
     // End of methods also called by native code.
     // Any new method exposed to native must be added after the last one, do not reorder
 
@@ -110,8 +112,6 @@ interface IAppOpsService {
     void startWatchingStarted(in int[] ops, IAppOpsStartedCallback callback);
     void stopWatchingStarted(IAppOpsStartedCallback callback);
 
-    void startWatchingModeWithFlags(int op, String packageName, int flags, IAppOpsCallback callback);
-
     void startWatchingNoted(in int[] ops, IAppOpsNotedCallback callback);
     void stopWatchingNoted(IAppOpsNotedCallback callback);
 
-- 
2.42.0.820.g83a721a137-goog

